lib/block-patterns.php

```diff
@@ -228,12 +209,16 @@ function() {
 			unregister_block_pattern( 'core/' . $core_block_pattern );
 		}
 
-		foreach ( $new_core_block_patterns as $core_block_pattern ) {
-			register_block_pattern(
-				'core/' . $core_block_pattern,
-				require __DIR__ . '/block-patterns/' . $core_block_pattern . '.php'
-			);
+		$request = new WP_REST_Request( 'GET', '/__experimental/pattern-directory/patterns' );
```

Comment:
Another issue we ran into over in WordPress.com was calling the endpoint on every request for the admin page. We decided to [guard the call](https://github.com/Automattic/wp-calypso/blob/94c1213c164e6e2e6f9d5ecd19ff769a5688303c/apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php#L175) behind a `wp_cache_get` check and then `wp_cache_set` the results from the API endpoint for a period of time so that each admin page request doesn't fire off an external API call.

Reply:
I would also recommend using some caching around `rest_do_request()` in this case, a transient makes sense to me. I don't consider this to be a blocker for merge though.

Reply:
Should we do the caching in the actual endpoint around the API call to pattern directory? I think we'd want to avoid this returning different data from the endpoint?